src/features/transactions/wrapEventSubscriber.ts

```diff
@@ -0,0 +1,54 @@
+import { Web3Provider } from "@ethersproject/providers";
+import { Dispatch } from "@reduxjs/toolkit";
+
+import { Contract } from "ethers";
+
+import { store } from "../../app/store";
+import {
+  decrementBalanceBy,
+  incrementBalanceBy,
+} from "../balances/balancesSlice";
+import { SubmittedTransaction } from "./transactionsSlice";
+
+const handleWrapEvent = (eventName: string, data: any, dispatch: any) => {
+  const transactions = store.getState().transactions;
+
+  let transaction: SubmittedTransaction | null =
+    transactions.all.find((t: any) => t.hash === data[2].transactionHash) ||
+    null;
+
+  // If we don't have a 'transaction', we don't already know about this swap
+  // and therefore don't need to update the UI.
+  if (!transaction) return;
+
+  dispatch(
+    decrementBalanceBy({
+      //@ts-ignore
+      tokenAddress: transaction.order.senderToken,
```

Comment:
A SubmittedTransaction is not supposed to have an order, I don't understand. Or is this something beyond the scope of this ticket?

Reply:
Yeah... this is a weird one that should have its own ticket. If you look at the reducer, most transactions will have orders, and this is where I find the data I need to update the balances. So either the SubmittedTransaction needs to be extended or we must do some other refactoring to avoid using ts-ignore all the time when dealing with this.

Reply:
I did it, let me know what you think

Reply:
Yeah, I don't think `SubmittedTransactionWithOrder` is even necessary. I assume all "succeeded" `SubmittedTransaction` should have an order. But I understand it's beyond the scope of this ticket.